src/ipahealthcheck/system/filesystemspace.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (C) 2019 FreeIPA Contributors see COPYING for license
+#
+
+import logging
+import shutil
+
+from ipahealthcheck.system.plugin import SystemPlugin, registry
+from ipahealthcheck.core.plugin import Result
+from ipahealthcheck.core import constants
+# from collections import namedtuple
+
+logger = logging.getLogger()
+
+
+@registry
+class FileSystemSpaceCheck(SystemPlugin):
+    """
+    """
+
+    # watch important directories for FreeIPA
+    _pathchecks = {
+        '/var/lib/dirsrv/': 1024,
+        '/var/log/': 1024,
+        '/var/log/audit/': 512,
+        '/var/tmp/': 512,
+        '/tmp': 512
+    }
+
+    # File systems reaching 90% capacity risk fragmentation.
+    # Defragmentation is never desirable and not available
+    # on ext4 anyway. So error out at 20% free space.
+    min_free_percent = 20
+
+    def get_fs_free_space(self, pathname):
+        stat = shutil.disk_usage(pathname)
+        return int(stat.free / 2**20)
+
+    def get_fs_free_space_percentage(self, pathname):
+        stat = shutil.disk_usage(pathname)
+        return int(stat.free * 100 / stat.total)
+
+    def check(self):
+        for store in self._pathchecks:
+            percent = self.get_fs_free_space_percentage(store)
+            if percent < self.min_free_percent:
+                yield Result(
+                    self, constants.ERROR,
+                    msg='%s: %s %s%% < %s%%' % (
+                        store, 'free space percentage under threshold:',
+                        percent, self.min_free_percent
```

Comment:
Add in the variables used to generate the data into msg as keywords, so store=store, percent=percent, min_free=self.min_free_percent.